source/includes/dev/_microservice.md

```diff
@@ -142,97 +301,225 @@ const logger = require('logger');
 
-#### Adding a new endpoint
+- We craft a JSON object with the content of the HTTP POST body
+- As this endpoint needs to make a call to another microservice (through Control Tower), we use [Nock](https://github.com/nock/nock) to mock that POST request to the `/v1/doc-datasets/json` endpoint. This way, your tests won't require actual running instances of Control Tower or other microservices to run.
+- We send the our previously crafted POST request to a running instance of our dataset microservice, along with a `loggedUser` spoof data.
+- We get the HTTP response, process it for easier handling, and proceed to validate that it's content is as expected.
 
-In this section we'll cover how you can add a new endpoint with new functionality to an existing microservice. The aim is not to be a comprehensive guide to cover all cases, but more of a quick entry point into day-to-day actions you may want to perform, which should be complemented by your own learning of how a microservice works - remember that all microservices, despite being structurally similar, have their own custom code and functionality.
+Different microservices and endpoint will have different requirements when it comes to testing, but the great majority of endpoints can be tested using simple variations of these steps. There are some additional considerations you should take into account when testing:
 
-To add a new endpoint, here's the short tasklist you have to tackle:
+- The example above creates an actual dataset, meaning a MongoDB (or equivalent mocks) need to exist. For MongoDB specifically, our approach so far has been to use a real MongoDB instance, and running the tests on a separate collection (´dataset-tests´ for example), aiming for isolation. Other microservices (for example, those relying on Elasticsearch) use mocks instead. Mocking usually leads to faster execution times, but can be troublesome to properly code. Use whichever alternative is best for you, and refer to the [Data layer](#data-layer) section for examples of microservices that use (and test with) different tools.
+- Nock has a [feature that blocks all HTTP requests](https://github.com/nock/nock#enabledisable-real-http-requests), which is useful to ensure your code or tests are not relying on an external service without you being aware - just be sure to whitelist your own IP, otherwise the HTTP call your test makes to your microservice will fail too.
+- Individual tests should run in isolation, and without assuming order. For example, running a test that first tests an insert, and then using the inserted element to test a delete would be a bad practice. Instead, your insert test should clean up its data once it's done, and the delete test should prepopulate the database before actually trying to delete it. A corollary of this is that you should be able to run your tests multiple times, back-to-back, without that affecting the results.
```

Comment:
I'd summarize this last bullet as "Always make sure your tests are idempotent." :grin: